ci/prepare.sh

```diff
@@ -0,0 +1,35 @@
+#!/bin/bash
+
+# Exit if anything fails AND echo each command before executing
+# http://www.peterbe.com/plog/set-ex
+set -ex
+
+if [ $# -lt 3 ]; then
+	echo "usage: $0 <db-name> <db-user> <db-pass> [db-host] [wp-version]"
+	exit 1
+fi
+
+# Set up constant
+DB_NAME=$1
+DB_USER=$2
+DB_PASS=$3
+DB_HOST=${4-localhost}
+WP_VERSION=${5-latest}
+
+WORDPRESS_SITE_DIR=${WORDPRESS_SITE_DIR-/tmp/wordpress}
+
+# Create WordPress root directory
+mkdir -p $WORDPRESS_SITE_DIR
+
+# Download WordPress
+vendor/bin/wp core download --force --version=$WP_VERSION --path=$WORDPRESS_SITE_DIR --allow-root
```

Comment:
I've not encountered any problems with WP-CLI installed via composer. That said the context I'm every use it in is for automated testing. In day-to-day use, I've had it installed globally. I'm happy to install WP-CLI globally if you think it'd be more reliable or just less confusing for users.

Regarding the `require-dev` - only if you checkout WordHat directly and run `install`. End users should be listing it has a dependency and running `install` on their own project - which case it will only install the `require` packages. (`--no-dev` is the flag you're thinking of, which only installs the current's projects non-dev dependencies).